cloudmarker/test/test_vmdatadiskencryptionevent.py

```diff
@@ -0,0 +1,153 @@
+        events = list(plugin.eval(record))
+        self.assertEqual(events, [])
+
+    def test_ext_bucket_missing(self):
+        record = copy.deepcopy(base_record)
+        record['ext'] = None
+        plugin = vmdatadiskencryptionevent.VMDataDiskEncryptionEvent()
+        events = list(plugin.eval(record))
+        self.assertEqual(events, [])
+
+    def test_instance_view_missing(self):
+        record = copy.deepcopy(base_record)
+        record['raw'] = copy.deepcopy(base_raw_bucket)
+        record['raw']['instance_view'] = None
+        record['ext']['all_data_disks_encrypted'] = False
+        plugin = vmdatadiskencryptionevent.VMDataDiskEncryptionEvent()
+        events = list(plugin.eval(record))
+        self.assertEqual(events, [])
+
+    def test_all_data_disks_encrypted(self):
+        record = copy.deepcopy(base_record)
+        plugin = vmdatadiskencryptionevent.VMDataDiskEncryptionEvent()
+        events = list(plugin.eval(record))
+        self.assertEqual(events, [])
+
+    def test_cloud_type_non_azure(self):
+        record = copy.deepcopy(base_record)
+        record['ext']['all_data_disks_encrypted'] = False
+        record['com']['cloud_type'] = 'non_azure'
+        plugin = vmdatadiskencryptionevent.VMDataDiskEncryptionEvent()
+        events = list(plugin.eval(record))
+        self.assertEqual(events, [])
+
+    def test_single_unencrypted_data_disk(self):
+        record = copy.deepcopy(base_record)
+        record['raw'] = copy.deepcopy(base_raw_bucket)
+        record['ext']['all_data_disks_encrypted'] = False
+        record['raw']['instance_view']['disks'].append(
+            copy.deepcopy(base_unencrypted_data_disk))
+        plugin = vmdatadiskencryptionevent.VMDataDiskEncryptionEvent()
+        events = list(plugin.eval(record))
+        self.assertEqual(len(events), 1)
+        self.assertEqual(events[0]['ext']['record_type'],
+                         'vm_data_disk_encryption_event')
+        self.assertEqual(events[0]['com']['record_type'],
+                         'vm_data_disk_encryption_event')
+
+    def test_single_encrypted_data_disk(self):
+        record = copy.deepcopy(base_record)
+        record['raw'] = copy.deepcopy(base_raw_bucket)
+        record['ext']['all_data_disks_encrypted'] = False
+        encrypted_data_disk = copy.deepcopy(base_unencrypted_data_disk)
+        encrypted_data_disk['encryption_settings'][0]['enabled'] = True
+        record['raw']['instance_view']['disks'].append(encrypted_data_disk)
+        plugin = vmdatadiskencryptionevent.VMDataDiskEncryptionEvent()
+        events = list(plugin.eval(record))
+        self.assertEqual(len(events), 0)
+
+    def test_multiple_unencrypted_data_disk(self):
+        record = copy.deepcopy(base_record)
+        record['raw'] = copy.deepcopy(base_raw_bucket)
+        record['ext']['all_data_disks_encrypted'] = False
+        unencrypted_data_disks = random.randint(1, 5)
```

Comment:
Why not have a definite number of disks here like say `3` or `5` instead of randomizing the number of disks?

Reply:
My intention was to introduce randomness to the test case. Suggestions?